src/coreclr/nativeaot/System.Private.CoreLib/src/System/Array.NativeAot.cs

```diff
@@ -555,72 +361,17 @@ private static unsafe void CopyImplPrimitiveTypeWithWidening(Array sourceArray,
             }
         }
 
-        public static unsafe void Clear(Array array)
+        [Intrinsic]
+        public unsafe int GetLength(int dimension)
         {
-            if (array == null)
-                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
+            int rank = this.GetMethodTable()->MultiDimensionalArrayRank;
```

Comment:
You have update this to match CoreCLR impl. Should it be moved to shared part as well?

Reply:
It is ok to pay the extra addition for non-zero lower bounds, even though they are not supported in native aot currently. 

If we want to avoid it, we can introduce a  `static bool IsNonZeroLowerBoundSupported` property on Array that return returns true on CoreCLR and false on NAOT, and use it to avoid the unnecessary addition.

Reply:
Sure, then GetLowerBound/GetUpperBound/GetLength can be unified too.